### DISCUSSION TITLE
Virtual Zones Expander

### DISCUSSION THREAD

**Discussion by AlFreddyTo** [taligentx/dscKeybusInterface#345] [dscKeybusInterface Discussion]
Hi, tks a lot for your work on this. It worked quite well for my DSC home alarm panel. I would like to know if the virtual zone expander is available, I am quite new in programming and on github as well, maybe it´s me but I cannot find any info on which libraries to use and how to wire the virtual zones. Thanks again !

**Comment by Dilbert66** [comment]
if you use Home Assistant or any non HA home control system that can use MQTT, you can try my ESPHome component which does offer virtual zone expander support as well a full keypad emulation. It's based on this library.  Note, that you will need to read up a bit on how to use ESPHome first. I can't assist with basic esphome/ha setups. I had provided some zone expander code here a long time ago but it never got integrated and the code has progressed quite a bit since then.
https://github.com/Dilbert66/esphome-dsckeybus/tree/dev

**Reply by AlFreddyTo** [reply]
Thanks very much for your answer. I have another question, is there any way of sending a "trigger alarm" command through dsc.write ?

Thanks again. Best regards
